src/tokenizedSubaccounts/BaseOnChainSigningTSA.sol

```diff
@@ -58,6 +58,21 @@ abstract contract BaseOnChainSigningTSA is BaseTSA {
     emit ActionSigned(msg.sender, hash, action);
   }
 
+  function signRfqActionData(IMatching.Action memory action, IRfqModule.TradeData[] memory trades)
```

Comment:
I would just change the `signActionData` to take in an extra bytes field like:

`function signActionData(IMatching.Action memory action, bytes memory extraData)`

Needs better naming than that, but wouldn't hardcode it as only RFQ

Then have `_verifyAction` take in the bytes, and logic for working out if the RFQ request is valid can live in PPTSA

You'll have to change CCTSA slightly, but a tiny change like this is not a concern

Reply:
Great idea. Let me do that instead.